File: standard-udfs-examples/standard-udfs-example-udfs-spark/src/test/scala/com/linkedin/stdudfs/examples/spark/TestMapFromTwoArraysFunctionWrapper.scala
/**
 * Copyright 2018 LinkedIn Corporation. All rights reserved.
 * Licensed under the BSD-2 Clause license.
 * See LICENSE in the project root for license information.
 */
package com.linkedin.stdudfs.examples.spark

import com.linkedin.stdudfs.spark.common.AssertSparkExpression._
import org.testng.annotations.{BeforeClass, Test}

import scala.collection.mutable.WrappedArray._

class TestMapFromTwoArraysFunctionWrapper {

  @BeforeClass
  def registerFunction(): Unit = {
    registerStandardUdf(
      "map_from_two_arrays",
      classOf[MapFromTwoArraysFunctionWrapper]
    )
  }

  @Test
  def testMapFromTwoArrays(): Unit = {
    assertFunction("map_from_two_arrays(array(1,2), array('a', 'b'))", Map((1, "a"), (2, "b")))

    assertFunction("map_from_two_arrays(array(array(1), array(2)), array(array('a'), array('b')))",
      Map((make(Array(1)), make(Array("a"))), (make(Array(2)), make(Array("b")))))

    assertFunction("map_from_two_arrays(null, array(array('a'), array('b')))", null)
    assertFunction("map_from_two_arrays(array(array(1), array(2)), null)", null)
  }
}

File: transportable-udfs-examples/transportable-udfs-example-udfs-avro/src/test/java/com/linkedin/transport/examples/avro/TestMapFromTwoArraysFunctionWrapper.java
/**
 * Copyright 2018 LinkedIn Corporation. All rights reserved.
 * Licensed under the BSD-2 Clause license.
 * See LICENSE in the project root for license information.
 */
package com.linkedin.transport.examples.avro;

import com.google.common.collect.ImmutableList;
import com.google.common.collect.ImmutableMap;
import org.apache.avro.Schema;
import org.testng.annotations.Test;

import static com.linkedin.transport.avro.common.AssertAvroUdf.*;
import static com.linkedin.transport.avro.common.SchemaFactory.*;


public class TestMapFromTwoArraysFunctionWrapper {
  @Test
  public void testMapFromTwoArrays() {
    assertFunction(new MapFromTwoArraysFunctionWrapper(), new Schema[]{array(STRING), array(STRING)},
        new Object[]{ImmutableList.of("1", "2"), ImmutableList.of("a", "b")}, ImmutableMap.of("1", "a", "2", "b"));

    assertFunction(new MapFromTwoArraysFunctionWrapper(), new Schema[]{array(STRING), array(array(STRING))},
        new Object[]{null, ImmutableList.of(ImmutableList.of("a"), ImmutableList.of("b"))}, null);

    assertFunction(new MapFromTwoArraysFunctionWrapper(), new Schema[]{array(STRING), array(array(STRING))},
        new Object[]{ImmutableList.of("1", "2"), null}, null);
  }
}
